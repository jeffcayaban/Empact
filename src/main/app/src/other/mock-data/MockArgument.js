import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};