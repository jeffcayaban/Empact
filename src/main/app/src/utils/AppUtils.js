import {API_BASE_URL, AUTHORIZATION_HEADER_FIELD} from '../constants';
import {ACCESS_TOKEN, AUTHORIZATION_TOKEN_TYPE, PETITION_LIST_SIZE} from "../constants";

/** Creates a request object and sends the request to the server. **/
const request = (options) => {
    const headers = new Headers({ 'Content-Type': 'application/json' });
    const accessToken = localStorage.getItem(ACCESS_TOKEN);

    if (accessToken !== undefined && accessToken !== null) {
        headers.append(AUTHORIZATION_HEADER_FIELD, `${AUTHORIZATION_TOKEN_TYPE} ${accessToken}`)
    }

    options = {...options, headers: headers};

    return fetch(options.url, options).then(response =>
            (response.json()).then(jsonResponse => (!response.ok) ? Promise.reject(jsonResponse) : jsonResponse)
    );
};

// ---------------------------- APIs ----------------------------

/** Gets all the petitions **/
export function getAllPetitions(page, size, sort, showClosed) {
    page = page || 0;
    size = size || PETITION_LIST_SIZE;

    return request({
        url: API_BASE_URL + "/petitions?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Gets a single petition given an ID **/
export function getPetition(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/" + petitionId,
        method: 'GET'
    });
}

/** Creates a new petition given its details **/
export function createPetition(petitionData) {
    return request({
        url: API_BASE_URL + "/petitions/create",
        method: 'POST',
        body: JSON.stringify(petitionData)
    });
}

/** Deletes an existing petition given its details **/
export function deletePetition(deletePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/delete",
        method: 'DELETE',
        body: JSON.stringify(deletePetitionData)
    });
}

/** Updates an existing petition with the given details **/
export function editPetition(editPetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/edit/",
        method: 'POST',
        body: JSON.stringify(editPetitionData)
    });
}

/** Logs in into a user account **/
export function login(loginRequest) {
    return request({
        url: API_BASE_URL + "/auth/signInUser",
        method: 'POST',
        body: JSON.stringify(loginRequest)
    });
}

/** Creates a new user account given the account details **/
export function signup(signupRequest) {
    return request({
        url: API_BASE_URL + "/auth/signUpUser",
        method: 'POST',
        body: JSON.stringify(signupRequest)
    });
}

/** Checks the username's availability given a username **/
export function checkUsernameAvailability(username) {
    return request({
        url: API_BASE_URL + "/user/checkUsernameAvailability?username=" + username,
        method: 'GET'
    });
}

/** Deletes a user **/
export function deleteUser(deleteUserRequest) {
    return request({
        url: API_BASE_URL + "/auth/deleteUser",
        method: 'DELETE',
        body: JSON.stringify(deleteUserRequest)
    })
}

/** Fetches the current user **/
export function getCurrentUser() {
    if(!localStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/getCurrentUser",
        method: 'GET'
    });
}

/** Fetches a user profile given a user's username **/
export function getUserProfile(username) {
    return request({
        url: API_BASE_URL + "/user/getUserProfile?username=" + username,
        method: 'GET'
    });
}

/** Fetches the petitions for a given user **/
export function getUserPetitions(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/user?username=" + username + "&page=" + page + "&size=" + size + "&sort=" + sort
        + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Fetches the most discussed petitions for a given user **/
export function getMostDiscussedPetitionsByUser(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitionsByUser?username=" + username + "&page=" + page +
            "&size=" + size + "&sort=" + sort
            + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Closes a petition given its details **/
export function closePetition(closePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/close",
        method: 'POST',
        body: JSON.stringify(closePetitionData)
    })
}

/** Creates a new argument by expert opinion given the new details **/
export function createArgumentByExpertOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Creates a new argument by popular opinion given the new details **/
export function createArgumentByPopularOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Fetches the arguments for a given petition **/
export function getArgumentsByRootPetitionId(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByRootPetitionId?rootPetitionId=" + petitionId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the arguments for a given argument **/
export function getArgumentsByArgumentId(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByArgumentId?argumentId=" + argumentId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the argument for a given ID **/
export function getArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/" + argumentId,
        method: 'GET'
    });
}

/** Fetches the parent argument for a given argument **/
export function getParentArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/getParentArgumentById?argumentId=" + argumentId,
        method: 'GET',
    });
}

/** Checks whether a petition is closed or not **/
export function isPetitionClosed(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/isPetitionClosed/" + petitionId,
        method: 'GET'
    })
}

/** Fetches the arguments for a given user **/
export function getUserArguments(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByUsernameAndAnonymity?username=" + username +
            "&showAnonymousContent=" + showAnonymousContent + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET'
    })
}

/** Fetches the number of supporting or opposing arguments for a given petition **/
export function countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countArgsByPetitionIdAndIsSupporting?petitionId=" + petitionId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Fetches the number of supporting or opposing arguments for a given argument **/
export function countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countAllByArgumentIdAndIsSupporting?argumentId=" + argumentId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Deletes an argument given its details **/
export function deleteArgument(deleteArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/delete",
        method: 'DELETE',
        body: JSON.stringify(deleteArgumentData)
    });
}

/** Updates an existing argument by popular opinion with new details **/
export function editArgumentByPopularOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Updates an existing argument by expert opinion with new details **/
export function editArgumentByExpertOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Fetches the most discussed petitions **/
export function getMostDiscussedPetitions(page, size, sort, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitions?page=" + page + "&size=" + size + "&sort=" + sort +
            "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments **/
export function getMostDiscussedArguments(pageSize, showClosed) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArguments?size=" + pageSize + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given petition **/
export function getMostDiscussedArgumentsForPetition(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForPetition?page=" + page + "&size=" + size + "&sort=" + sort +
            "&petitionId=" + petitionId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given argument **/
export function getMostDiscussedArgumentsForArgument(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForArgument?page=" + page + "&size=" + size + "&sort=" + sort +
            "&argumentId=" + argumentId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given user **/
export function getMostDiscussedArgumentsForUser(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForUser?page=" + page + "&size=" + size + "&sort=" + sort +
            "&username=" + username + "&showAnonymousContent=" + showAnonymousContent,
        method: 'GET'
    });
}

/** Fetches the data nodes required for the data visualisation of arguments **/
export function getDataNodes(contentId, isPetition) {
    return request({
        url: API_BASE_URL + "/arguments/getDataNodes?contentId=" + contentId + "&isPetition=" + isPetition,
        method: 'GET'
    });
}

/** Fetches data required for the petition report **/
export function getPetitionReport(petitionId) {
    return request({
        url: API_BASE_URL + "/arguments/getPetitionReport?petitionId=" + petitionId,
        method: 'GET'
    });
}

/** Searches for petitions given a query **/
export function searchPetitions(page, size, sort, showClosed, query) {
    return request({
        url: API_BASE_URL + "/petitions/search?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed
         + "&query=" + query,
        method: 'GET'
    });
}
