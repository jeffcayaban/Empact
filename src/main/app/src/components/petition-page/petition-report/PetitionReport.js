import React from "react";
import { Button, Modal } from "react-bootstrap";
import {getPetitionReport} from "../../../utils/AppUtils";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faCheck, faTimes} from "@fortawesome/free-solid-svg-icons";
import ArgumentHorizontalList from "./ArgumentHorizontalList";

/**
 * Displays a summary of the petition based on its arguments. It highlights whether the majority has supported or
 * opposed the petition, in addition to displaying the arguments that have agreed or disagreed with certain elements
 * of the petition.
 */

class PetitionReport extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            isPetitionWinning: false,
            supportingSituation: [],
            supportingActionGoal: [],
            notSupportingSituation: [],
            notSupportingActionGoal: []
        }
    }

    // Fetches the data for the petition report.
    fetchPetitionReport(petitionId) {
        getPetitionReport(petitionId)
            .then(response => {
                this.setState({
                    isPetitionWinning: response.isPetitionWinning,
                    supportingSituation: response.supportingSituation,
                    supportingActionGoal: response.supportingActionGoal,
                    notSupportingSituation: response.notSupportingSituation,
                    notSupportingActionGoal: response.notSupportingActionGoal
                })
            })
            .catch(error => console.error(error));
    }

    // Generates a message regarding the petition's situation.
    static generateArgsSituationMessage(userArguments, isSupporting) {
        if (userArguments.length > 0) {
            // Assumes that there exists arguments.

            const isSupportingText = isSupporting ? "supports" : "does not support";
            return (
                <div>
                    <p>
                        There has been {userArguments.length} arguments that {isSupportingText} the <strong>situation
                        being true</strong>:
                    </p>
                    <div style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                        <ArgumentHorizontalList userArguments={userArguments} />
                    </div>
                </div>
            )
        }
    }

    // Generates a message regarding the petition's action.
    static generateArgsSituationActionMessage(userArguments, isSupporting) {
        if (userArguments.length > 0) {
            // Assumes that there exists arguments.

            const isSupportingText = isSupporting ? "supports" : "does not support";
            return (
                <div>
                    <p>
                        There has been { userArguments.length } arguments that {isSupportingText} the
                        <strong> proposed action in achieving the goal</strong>:
                    </p>
                    <div style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                        <ArgumentHorizontalList userArguments={userArguments} />
                    </div>
                </div>
            )
        }
    }

    // Checks whether there exists any arguments.
    existAnyArguments() {
        const {supportingSituation, supportingActionGoal, notSupportingSituation, notSupportingActionGoal} = this.state;
        return (
            (supportingSituation !== undefined && supportingSituation.length > 0) ||
            (supportingActionGoal !== undefined && supportingActionGoal.length > 0) ||
            (notSupportingSituation !== undefined && notSupportingSituation.length > 0) ||
            (notSupportingActionGoal !== undefined && notSupportingActionGoal.length > 0)
        )
    }

    componentDidMount() {
        this.fetchPetitionReport(this.props.petition.id);
    }

    render() {
        const { showReport, handleShowPetitionReport } = this.props;
        const { isPetitionWinning, supportingSituation, supportingActionGoal, notSupportingSituation,
            notSupportingActionGoal } = this.state;
        const isWinning = isPetitionWinning ? "supported" : "opposed";

        let petitionReportContent;

        if (this.existAnyArguments()) {
            // Assumes that there exists arguments for the petition.

            petitionReportContent = (
                <div>
                    <div className={"centerStyle"}>
                        <h3 align="center">{`The majority has `} <strong>{isWinning}</strong> {`your petition`}</h3>
                    </div>
                    <div className={"titleContentPadding"}>
                        <div>
                            <div>
                                <FontAwesomeIcon icon={faCheck} style={{ color: 'green' }}/>
                                <span style={{ fontSize: '13pt', margin: 0, paddingLeft: '10px' }}>What is supported:</span>
                            </div>
                            <hr style={{ marginTop: '10px' }} />
                            { PetitionReport.generateArgsSituationMessage(supportingSituation, true) }
                            { PetitionReport.generateArgsSituationActionMessage(supportingActionGoal, true) }
                        </div>

                        <div style={{ paddingTop: '30px' }}>
                            <div>
                                <FontAwesomeIcon icon={faTimes} style={{ color: 'red' }}/>
                                <span style={{ fontSize: '13pt', margin: 0, paddingLeft: '10px' }}>What is opposed:</span>
                            </div>
                            <hr style={{ marginTop: '10px' }}/>
                            { PetitionReport.generateArgsSituationMessage(notSupportingSituation, false) }
                            { PetitionReport.generateArgsSituationActionMessage(notSupportingActionGoal, false) }
                        </div>
                    </div>
                </div>
            )
        } else {
            // Assumes that there exists no arguments for the petition.

            petitionReportContent = (
                <div>
                    <div className={"centerStyle"}>
                        <h3 align="center">No arguments has been made to this petition.</h3>
                    </div>
                </div>
            )
        }

        return (
            <Modal show={showReport} bsSize="large">
                <Modal.Header>
                    <Modal.Title>Petition Report</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    { petitionReportContent }
                </Modal.Body>
                <Modal.Footer>
                    <Button onClick={handleShowPetitionReport}>
                        Close
                    </Button>
                </Modal.Footer>
            </Modal>
        );
    }
}

export default PetitionReport;