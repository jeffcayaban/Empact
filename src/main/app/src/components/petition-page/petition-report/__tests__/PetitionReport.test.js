import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import * as AppUtils from '../../../../utils/AppUtils';
import MockArgument from '../../../../other/mock-data/MockArgument';
import PetitionReport from "../PetitionReport";

describe('PetitionReport', () => {

    it('should display a no arguments message', () => {
        const component = shallow(<PetitionReport
            showReport={true}
            handleShowPetitionReport={jest.fn()}
            petition={{ id: 'testId' }}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully render the page', () => {
        const originalGetPetitionReport = AppUtils.getPetitionReport;
        const petitionReportResponse = {
            supportingSituation: [MockArgument],
            supportingActionGoal: [MockArgument],
            notSupportingSituation: [MockArgument],
            notSupportingActionGoal: [MockArgument]
        };

        AppUtils.getPetitionReport = () => Promise.resolve(petitionReportResponse);

        const component = shallow(<PetitionReport
            showReport={true}
            handleShowPetitionReport={jest.fn()}
            petition={{ id: 'testId' }}
        />);

        component.setState(petitionReportResponse);

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.getPetitionReport = originalGetPetitionReport;
    });

});
