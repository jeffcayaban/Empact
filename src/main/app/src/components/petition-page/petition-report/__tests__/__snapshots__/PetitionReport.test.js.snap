// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetitionReport should display a no arguments message 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  bsSize="large"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Petition Report
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      <div
        className="centerStyle"
      >
        <h3
          align="center"
        >
          No arguments has been made to this petition.
        </h3>
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;

exports[`PetitionReport should successfully render the page 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  bsSize="large"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Petition Report
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div>
      <div
        className="centerStyle"
      >
        <h3
          align="center"
        >
          The majority has 
           
          <strong>
            opposed
          </strong>
           
          your petition
        </h3>
      </div>
      <div
        className="titleContentPadding"
      >
        <div>
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f00c",
                    "M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z",
                  ],
                  "iconName": "check",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "green",
                }
              }
              symbol={false}
              title=""
              transform={null}
            />
            <span
              style={
                Object {
                  "fontSize": "13pt",
                  "margin": 0,
                  "paddingLeft": "10px",
                }
              }
            >
              What is supported:
            </span>
          </div>
          <hr
            style={
              Object {
                "marginTop": "10px",
              }
            }
          />
          <div>
            <p>
              There has been 
              1
               arguments that 
              supports
               the 
              <strong>
                situation being true
              </strong>
              :
            </p>
            <div
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <ArgumentHorizontalList
                userArguments={
                  Array [
                    Object {
                      "agreesWithCQ": true,
                      "argumentType": "ExpertOpinion",
                      "createdBy": Object {
                        "authorities": Array [
                          "ROLE_ADMIN",
                        ],
                        "firstName": "testFirstName",
                        "id": "testId",
                        "lastName": "testLastName",
                        "username": "testUsername",
                      },
                      "creationDateTime": 1551716401541,
                      "criticalQuestionId": "CQ1",
                      "expert": "testExpert",
                      "expertAssertion": "testAssertion",
                      "expertDomain": "testDomain",
                      "id": "testId",
                      "isAnonymous": false,
                      "isSupporting": true,
                      "lastUpdatedDateTime": 1551716402000,
                      "parentArgumentId": "testParentArgumentId",
                      "rootPetitionId": "testRootPetitionId",
                      "sources": Array [],
                      "subCriticalQuestionId": null,
                    },
                  ]
                }
              />
            </div>
          </div>
          <div>
            <p>
              There has been 
              1
               arguments that 
              supports
               the
              <strong>
                 proposed action in achieving the goal
              </strong>
              :
            </p>
            <div
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <ArgumentHorizontalList
                userArguments={
                  Array [
                    Object {
                      "agreesWithCQ": true,
                      "argumentType": "ExpertOpinion",
                      "createdBy": Object {
                        "authorities": Array [
                          "ROLE_ADMIN",
                        ],
                        "firstName": "testFirstName",
                        "id": "testId",
                        "lastName": "testLastName",
                        "username": "testUsername",
                      },
                      "creationDateTime": 1551716401541,
                      "criticalQuestionId": "CQ1",
                      "expert": "testExpert",
                      "expertAssertion": "testAssertion",
                      "expertDomain": "testDomain",
                      "id": "testId",
                      "isAnonymous": false,
                      "isSupporting": true,
                      "lastUpdatedDateTime": 1551716402000,
                      "parentArgumentId": "testParentArgumentId",
                      "rootPetitionId": "testRootPetitionId",
                      "sources": Array [],
                      "subCriticalQuestionId": null,
                    },
                  ]
                }
              />
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "paddingTop": "30px",
            }
          }
        >
          <div>
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    352,
                    512,
                    Array [],
                    "f00d",
                    "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                  ],
                  "iconName": "times",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              style={
                Object {
                  "color": "red",
                }
              }
              symbol={false}
              title=""
              transform={null}
            />
            <span
              style={
                Object {
                  "fontSize": "13pt",
                  "margin": 0,
                  "paddingLeft": "10px",
                }
              }
            >
              What is opposed:
            </span>
          </div>
          <hr
            style={
              Object {
                "marginTop": "10px",
              }
            }
          />
          <div>
            <p>
              There has been 
              1
               arguments that 
              does not support
               the 
              <strong>
                situation being true
              </strong>
              :
            </p>
            <div
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <ArgumentHorizontalList
                userArguments={
                  Array [
                    Object {
                      "agreesWithCQ": true,
                      "argumentType": "ExpertOpinion",
                      "createdBy": Object {
                        "authorities": Array [
                          "ROLE_ADMIN",
                        ],
                        "firstName": "testFirstName",
                        "id": "testId",
                        "lastName": "testLastName",
                        "username": "testUsername",
                      },
                      "creationDateTime": 1551716401541,
                      "criticalQuestionId": "CQ1",
                      "expert": "testExpert",
                      "expertAssertion": "testAssertion",
                      "expertDomain": "testDomain",
                      "id": "testId",
                      "isAnonymous": false,
                      "isSupporting": true,
                      "lastUpdatedDateTime": 1551716402000,
                      "parentArgumentId": "testParentArgumentId",
                      "rootPetitionId": "testRootPetitionId",
                      "sources": Array [],
                      "subCriticalQuestionId": null,
                    },
                  ]
                }
              />
            </div>
          </div>
          <div>
            <p>
              There has been 
              1
               arguments that 
              does not support
               the
              <strong>
                 proposed action in achieving the goal
              </strong>
              :
            </p>
            <div
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <ArgumentHorizontalList
                userArguments={
                  Array [
                    Object {
                      "agreesWithCQ": true,
                      "argumentType": "ExpertOpinion",
                      "createdBy": Object {
                        "authorities": Array [
                          "ROLE_ADMIN",
                        ],
                        "firstName": "testFirstName",
                        "id": "testId",
                        "lastName": "testLastName",
                        "username": "testUsername",
                      },
                      "creationDateTime": 1551716401541,
                      "criticalQuestionId": "CQ1",
                      "expert": "testExpert",
                      "expertAssertion": "testAssertion",
                      "expertDomain": "testDomain",
                      "id": "testId",
                      "isAnonymous": false,
                      "isSupporting": true,
                      "lastUpdatedDateTime": 1551716402000,
                      "parentArgumentId": "testParentArgumentId",
                      "rootPetitionId": "testRootPetitionId",
                      "sources": Array [],
                      "subCriticalQuestionId": null,
                    },
                  ]
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
